From e79ec2ffde1f45a0cd679ce20fc63edc2589e3c3 Mon Sep 17 00:00:00 2001
From: Kiran Kelageri <kirankelageri@codeaurora.org>
Date: Wed, 22 Aug 2018 18:35:31 -0700
Subject: [PATCH 05/12] Bluetooth-TWS: UI updates based on profile connection.

Changes are made as per aosp implementation for TWS,
UI update is moved to profile connection states instead
of depending connection states.

CRs-fixed: 2300825
Change-Id: I7074698b3c421822756c30a808da6cd02968bced
---
 .../bluetooth/SavedBluetoothTwsDeviceUpdater.java         | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/com/android/settings/bluetooth/SavedBluetoothTwsDeviceUpdater.java b/src/com/android/settings/bluetooth/SavedBluetoothTwsDeviceUpdater.java
index 8759e6f..4504318 100644
--- a/src/com/android/settings/bluetooth/SavedBluetoothTwsDeviceUpdater.java
+++ b/src/com/android/settings/bluetooth/SavedBluetoothTwsDeviceUpdater.java
@@ -50,6 +50,7 @@ package com.android.settings.bluetooth;
 
 import android.bluetooth.BluetoothAdapter;
 import android.bluetooth.BluetoothDevice;
+import android.bluetooth.BluetoothProfile;
 import android.content.Context;
 
 import com.android.settings.connecteddevice.DevicePreferenceCallback;
@@ -74,11 +75,12 @@ public class SavedBluetoothTwsDeviceUpdater extends BluetoothDeviceUpdater {
     }
 
     @Override
-    public void onConnectionStateChanged(CachedBluetoothDevice cachedDevice, int state) {
+    public void onProfileConnectionStateChanged(CachedBluetoothDevice cachedDevice, int state,
+            int bluetoothProfile) {
         final BluetoothDevice device = cachedDevice.getDevice();
-        if (state == BluetoothAdapter.STATE_CONNECTED) {
+        if (state == BluetoothProfile.STATE_CONNECTED) {
             removePreference(cachedDevice);
-        } else if (state == BluetoothAdapter.STATE_DISCONNECTED &&  device.isTwsPlusDevice()) {
+        } else if (state == BluetoothProfile.STATE_DISCONNECTED && device.isTwsPlusDevice()) {
             addPreference(cachedDevice);
         }
     }
-- 
2.28.0

